### ISSUE TITLE
As a user, I want to be able to edit open revisions via the tasklist, so that I can record changes to revisions

### ISSUE THREAD

**Issue by BCerki** [bcgov/cas-cif#1561]
#### Description:

Related to https://github.com/bcgov/cas-cif/issues/1284. 1284 describes a different edit workflow that we may additionally implement, but for now the only edit pathway with be via this ticket (via the tasklist).

Prototype: [Edit Revision - Update diff ("Forms updated" section)](https://www.figma.com/proto/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?page-id=2289%3A2663&node-id=7336-362&viewport=-470%2C1174%2C0.2&scaling=scale-down&starting-point-node-id=7336%3A362&show-proto-sidebar=1)

#### Acceptance Criteria:

Given I am a user and I navigate to any of the forms
When I view the form
Then I do not see an edit button

Given I am a user and I navigate away from a revision and return to it (by clicking "View/Edit")
When I access any of the forms via the tasklist
Then these forms are in edit mode

Given I am a user and I submit my revision
When I access any of the forms via the tasklist
Then these forms are back to view mode

#### Development Checklist:

- [x] Remove the edit button from the forms
- [x] `onClick` of a revision's "View/Edit" button, set the tasklist mode to "edit"
- [x] `onSubmit` of a revision should behave the same as it currently does, submit the project and redirect to the projects list
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by BCerki** [comment]
@nanyangpro , could you please add the link to the most recent wireframe?

**Comment by BCerki** [comment]
Potential for good first issue, however would like to get others' opinions via estimation before tagging

**Comment by pbastia** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/1561?planning-poker) with Zenhub @BCerki @dleard @gurjmatharu @sam-warren @Sepehr-Sobhani @tmastrom

**Comment by Dianadec** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/1561?planning-poker) with Zenhub @BCerki @dleard @gurjmatharu @pbastia @sam-warren @Sepehr-Sobhani @tmastrom @mikevespi

**Comment by pbastia** [comment]
Description of work done and TBD: 


	
app/pages/cif/project-revision/[projectRevision]/form/[formIndex]/index.tsx used to hold the logic for setting the tasklist's mode to edit when the edit button is pressed. I removed the logic that does so as well as the piece of code that renders the button on the forms.
	
app/components/ProjectRevision/ProjectRevisionTableRow.tsx is where the logic should live to tell the tasklist component to go into edit mode when the "View/Edit" button is pressed on the Amendments & Revisions page. I have included some properties to the fragment in this file to call useFormIndexHelpers to utilize functions handleResumeRevision and handleCreateRevision when the "View/Edit" button is pressed. 
	
app/hooks/useRedirectToValidFormIndex.ts is the hook that tells the router to redirect to a valid form index if none is provided. I fixed a typo in this file (see line 23) and updated the dependency array to include router.asPath. Currently this file is causing an issue that sends the amendments/revisions page into an infinite useEffect loop when the "View/Edit" button is pressed. I have a suspicion it has to do with the useEffect hook in this file.
	
Work left to do: Fix the infinite useEffect loop and set the tasklist mode to edit on click of the "View/Edit" button.
